sectionproperties/pre/sections.py

```diff
@@ -1500,7 +1500,10 @@ def offset_perimeter(self, amount: float = 0, where = "exterior", resolution: fl
             for geom in self.geoms:
                 # Use symmetric intersection to find the region of the original
                 # that intersects with the eroded unionized shape
-                geoms_acc.append(geom & offset_geom) 
+                try:
+                    geoms_acc.append(geom & offset_geom)
+                except ValueError:
+                    raise ValueError(f"Try a smaller offset. Section as completely eroded away...")
```

Comment:
While I think this works, there may be some cases where the user may want to model the complete erosion and continue the analysis e.g. a stiffening plate completely corrodes.

Would it be possible to modify the above to allow this?

Reply:
@robbievanleeuwen To clarify, the eroding of the CompoundGeometry erodes the whole section. It does not allow for the eroding of one part of the compound geometry, like a top plate on some section. That would be pretty nifty and I think could be implemented (ie, erode the geometry with index i) but the ValueError catches, I believe, when any of the geometries erode away. Ah, I think I see what you mean. Will investigate and see what I can come up with. 

Also, do you think it would be worthwhile to add the "index" parameter, to offset just one of the geometries within the compund?

Reply:
Agreed that would be a good feature, if possible, but above an beyond what I was thinking. This feature could be useful if certain sides were protected against corrosion while others exposed (a common design scenario).

It seems you might have clued onto what I was thinking about, here's a quick unchecked example which throws an error eventually:

```python
import sectionproperties.pre.sections as sections
from sectionproperties.analysis.cross_section import Section


uc = sections.i_section(d=400,b=400,t_f=25,t_w=25,r=30,n_r=8)
plate1 = sections.rectangular_section(b=500,d=10).align_center(uc).align_to(uc,"top")
plate2 = sections.rectangular_section(b=500,d=10).align_center(uc).align_to(uc,"bottom")
geom = uc + plate1 + plate2

for i in range(12):
    new_geom = geom.offset_perimeter(-i)
    new_geom.create_mesh([100])
    section = Section(new_geom)
    section.plot_mesh()
    # calculate properties etc...
```

Reply:
The error is generated when the section is eroded to the point it becomes a LineString and LineStrings cannot be handled by sectionproperties as an input geometry which is what is causing the error. It's a special case that needs to be filtered out.

Reply:
Do you see anything won't with just discarding the LineStrings once they've completely eroded? E.g. remove the `except` but keep the `try`?